### PR TITLE
mgr/cephadm: make client-keyring deploying ceph.conf optional

### PR DESCRIPTION
**Pull request by adk3798**
There are cases where users would like to manage their own
    ceph.conf but still have cephadm deploy the client keyrings,
    so this is being added to facilitate that.
    
    Fixes: https://tracker.ceph.com/issues/65335
    
    Signed-off-by: Adam King <adking@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/cephadm/operations.rst [hunk 1]
```diff
@@ -601,6 +601,13 @@ The resulting keyring file is:
 
   -rw-r-----. 1 qemu qemu 156 Apr 21 08:47 /etc/ceph/client.client.rbd.keyring
 
+By default, cephadm will also manage /etc/ceph/ceph.conf on hosts where it writes the keyrings.
```
Comment (anthonyeleven) [inline doc/cephadm/operations.rst:604]: Suggest backticks around the filename

File: doc/cephadm/operations.rst [hunk 2]
```diff
@@ -601,6 +601,13 @@ The resulting keyring file is:
...
+This aspect of the feature can be avoided by passing ``--no-ceph-conf`` when setting the keyring.
```
Comment (anthonyeleven) [inline doc/cephadm/operations.rst:605]: s/aspect of the//
s/avoided/suppressed/

**Comment by anthonyeleven** [comment]
LGTM.  Since there's code here I'm not qualified to approve the PR overall.

**Review by phlogistonjohn — APPROVED** [review]
looks ok to me

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-04-08_12:15:10-orch:cephadm-wip-adk-testing-2024-04-04-1735-distro-default-smithi/

failures were all either `in cluster log` stuff or mds_upgrade_sequence (both known issues) except one instance of https://tracker.ceph.com/issues/65155 because I forgot to tag the PR that fixes it.

Nothing to block merging

**Pull request merged**